tests/director/tests/application_api_test.go

```diff
@@ -705,23 +699,25 @@ func TestUnpairApplication(t *testing.T) {
 		defaultValue := "DEFAULT"
 		scenarios := []string{defaultValue, "test-scenario"}
 
+		fixtures.UpsertScenariosLabelDefinitionWithinTenant(t, ctx, certSecuredGraphQLClient, tenantID, scenarios)
+		defer fixtures.UpdateScenariosLabelDefinitionWithinTenant(t, ctx, certSecuredGraphQLClient, tenantID, []string{defaultValue})
+
 		applicationInput := fixtures.FixSampleApplicationRegisterInput("first")
-		applicationInput.Labels = graphql.Labels{ScenariosLabel: scenarios}
+		applicationInput.Labels = graphql.Labels{ScenariosLabel: scenarios, conf.ApplicationTypeLabelKey: createAppTemplateName("Cloud for Customer")}
 		appInputGQL, err := testctx.Tc.Graphqlizer.ApplicationRegisterInputToGQL(applicationInput)
 		require.NoError(t, err)
 
 		createApplicationReq := fixtures.FixRegisterApplicationRequest(appInputGQL)
 		application := graphql.ApplicationExt{}
 
 		err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tenantID, createApplicationReq, &application)
+		require.NoError(t, err)
```

Comment:
Think it's better first to execute the defers and then check for errors so we would be sure that the defers will be executed

Reply:
If we defer before checking for errors the evaluation of `formation.FormationTemplateID` will cause a nil ptr and we'll lose the error message on the following line.